WooFoundation/WooFoundation/Extensions/Date+StartAndEnd.swift

```diff
@@ -96,6 +96,47 @@ public extension Date {
         return calendar.date(byAdding: components, to: startOfMonth)
     }
 
+    // MARK: Quarter
+
+    /// Returns self's start of quarter in the given time zone.
+    func startOfQuarter(timezone: TimeZone, calendar: Calendar) -> Date? {
+        guard let startOfMonth = startOfMonth(timezone: timezone) else {
+            return nil
+        }
+
+        var components = calendar.dateComponents([.month, .year], from: startOfMonth)
+        switch components.month {
+        case 1, 2, 3:
+            components.month = 1
+        case 4, 5, 6:
+            components.month = 4
+        case 7, 8, 9:
+            components.month = 7
+        case 10, 11, 12:
+            components.month = 10
+        default:
+            return nil
+        }
+
+        return calendar.date(from: components)
+    }
+
+    /// Returns self's end of quarter in the given time zone.
+    func endOfQuarter(timezone: TimeZone, calendar: Calendar) -> Date? {
+        guard let startOfQuarter = startOfQuarter(timezone: timezone, calendar: calendar) else {
+            return nil
+        }
+
+        guard let endOfQuarter = calendar.date(byAdding: .month, value: 2, to: startOfQuarter) else {
+            return nil
+        }
+
+        var oneMonthUnit = DateComponents()
+        oneMonthUnit.month = 1
+        oneMonthUnit.second = -1
+        return calendar.date(byAdding: oneMonthUnit, to: endOfQuarter)
```

Comment:
The `endOfQuarter` constant is a little confusing here, since it isn't really the end of the quarter. Could we do the calculation by directly adding a quarter (3 months) to the start of the quarter, instead? 

For example:

```suggestion
        var oneQuarterUnit = DateComponents()
        oneQuarterUnit.month = 3
        oneQuarterUnit.second = -1
        return calendar.date(byAdding: oneQuarterUnit, to: startOfQuarter)
```

Reply:
This makes waaaay more sense 😅. Thanks for the suggestion! Fixed here: https://github.com/woocommerce/woocommerce-ios/pull/8325/commits/a0db47a07edc6d380714189cd809be4f9b8992ee